SmplEditor/MainWindow.xaml.cs

```diff
@@ -127,24 +130,34 @@ private void OnImportPlaylistClicked(object sender, RoutedEventArgs e)
                 else{ // existing song found on library
                     if(!matched.HasSmplSong()){ // if the song doesn't have SMPLSong, add the new information
                         matched.SmplMusic = targetSong;
+                        existingiTunesSongs.Add(matched);
                     }
                 }
                 remappedPlaylist.Add(matched);
             }
             return remappedPlaylist;
         }
-        private List<Song> addNewSongsToLibrary(ITunesPlaylist playlist, List<Song> library){
-            List<Song> newSongsList = new List<Song>();
-            return newSongsList;
-        }
-        private Playlist getPlaylist(Smpl playlist, List<Song> library){
-            Playlist newPlaylist = new Playlist();
-            return newPlaylist;
-        }
-        private Playlist getPlaylist(ITunesPlaylist playlist){
-            Playlist newPlaylist = new Playlist();
-            return newPlaylist;
+
+        private Dictionary<ITunesLibraryParser.Track, Song> linkTracksToLibrary(List<ITunesLibraryParser.Track> importedTracks, List<Song> library, List<Song> newSongs, List<Song> existingSmplSongs){
```

Comment:
Add descriptions with reasoning of why this method returns a dictionary.
Unlike the SMPL implementation of the overload with the similar feature, for iTunes implementation we do not use each playlists for the tracks. We use the already arranged duplicate-less list of tracks the iTunes provides.
To justify this behavior, lets compare 1. per playlist linking and 2. bulk linking.
Method 1. is how it is done on SMPL overload. Each track is compared to the combined songLibrary.
Method 2. is how this iTunes overload work. This method creates a mapping from the importing list to the songLibrary. In the importing method, each track is compared (or looked up) to the dictionary and mapped to the songLibrary, Instead of having to be compared to the whole songLibrary.
The gain is minimal if the implementation of dict_lookup is same as list_find, value comparison. But if it is hash-based search, this should eliminate huge overhead of track matching for duplicate tracks across playlists.

A bigger factor to consider may be the code readability.
Because this implementation creates different return type between overloads (are they still called overloads if the return types are different?), it actually harms the readability. But I would argue it is just a slight bit because the Playlist constructor requires similar parameters for both, the SMPL version and the iTunes version.